distributed/scheduler.py

```diff
@@ -7735,6 +7735,14 @@ def check_idle(self) -> int | None:
             self.idle_since = time()
             return self.idle_since
 
+        if self.jupyter:
+            last_activity = (
+                self._jupyter_server_application.web_app.last_activity().timestamp()
+            )
+            if last_activity > self.idle_since:
+                self.idle_since = last_activity
+                return self.idle_since
```

Comment:
It feels to me like it would be better not to skip idle timeout check in this case.

In practice it's probably fine to skip. With (eg) 2 hour idle timeout, `check_idle` should be called every 30 minutes, so if there's Jupyter activity since that last check it also means there's been activity that's recent enough so there won't be idle timeout on this check. But that required some thought, and it seems to me better if the logic around idle timeout isn't more complicated / doesn't require much thought / doesn't break if `check_idle` were to be called less frequently.

(It's also possible I'm not reasoning correctly about the logic here.)

Reply:
We do need to return `idle_since` in this case, because otherwise if we continue on to the idle check, we'll always return `None`. If we don't return `idle_since` here, then we'll _never_ return a last activity time from Jupyter for this function.

I think this design of returning `None` is a bit odd, but any refactoring there can be done separately; see https://github.com/dask/distributed/pull/7687#discussion_r1145095196.

All the logic here would be simpler if we just stored, and always returned, a `Scheduler.last_activity` timestamp, like how Jupyter does. The `None` value makes things more complex IMO.

Reply:
Yeah I agree. I think the naming of `idle_since` and `check_idle()` lends itself to this problem because the scheduler is often not idle, so what should those values be if the scheduler isn't idle? Currently, if the scheduler is not idle it returns `None` to signify that it is not idle, which is not intuitive.

Switching to `last_activity` sounds more sensible because if the scheduler is not idle this would just be `time.time()`.

Reply:
Agreed. Maybe switching to `last_activity` can be a follow-up PR.